modules/core/src/main/java/org/apache/ignite/internal/util/ArrayUtils.java

```diff
@@ -295,7 +298,22 @@ public static boolean nullOrEmpty(boolean[] arr) {
     public static <T> List<T> asList(@Nullable T... vals) {
         return nullOrEmpty(vals) ? Collections.emptyList() : Arrays.asList(vals);
     }
-
+    
+    /**
+     * Converts array to {@link Set}.
+     *
+     * <p>Note that unlike {@link Arrays#asList(Object[])}, this method is {@code null}-safe. If {@code null} is passed in, then empty set
```

Comment:
Also it may worth to note, the resulted Set is mutable in contrast to Set.of(T...).
Does mutability matter?

Reply:
Streams are not GC friendly.
Let's rewrite to for-loop or Collections.addAll(new HashSet(), vals)

Reply:
got it !